book/plugins/how-to-guides/maximize-compatibility.md

```diff
@@ -18,13 +18,259 @@ Some guidance on each of these approaches:
+
+You'll follow the same steps above for creating an `environment-files/` directory and for the naming structure of your environment file, but the contents of your environment file(s) will look a bit different.
+We'll explore an example plugin below that requires q2-feature-table and q2-composition.
+
+```
+channels:
+- https://packages.qiime2.org/qiime2/2024.5/tiny/released
+- https://packages.qiime2.org/qiime2/2024.5/amplicon/released
+- conda-forge
+- bioconda
+- defaults
+dependencies:
+  - qiime2-tiny
+  - q2-feature-table
+  - q2-composition
+  - pip
+  - pip:
+    - my_plugin@git+https://github.com/my-plugin-org/my-plugin.git@release-2024.5
+```
+
+In this example, the plugin being developed requires q2-feature-table and q2-composition but is not compatible with the entire amplicon distribution.
+In order for this plugin to still work with a basic QIIME 2 environment, the tiny distribution must be included in addition to the two QIIME 2 plugin dependencies.
+The tiny distribution's metapackage will be installed from the first channel that includes tiny, and because the two plugin dependencies are a part of the amplicon distribution, this is why the second conda channel provided is from amplicon.
+
+As a general rule, this will be the structure of your customized environment files:
+```
+channels:
+- https://packages.qiime2.org/qiime2/20XX.REL/tiny/released
+- https://packages.qiime2.org/qiime2/20XX.REL/<my-plugin-deps-distro>/released
+- conda-forge
+- bioconda
+- defaults
+dependencies:
+  - qiime2-tiny
+  - <my-plugin-dep-1>
+  - pip
+  - pip:
+    - my_plugin@git+https://github.com/my-plugin-org/my-plugin.git@release-2024.5
+```
+
+With <my-plugin-deps-distro> corresponding to the distribution where the QIIME 2 dependencies for your plugin live, and <my-plugin-dep-1> being the QIIME 2 dependenc(ies) needed.
+Note that if you have plugin dependencies spanning multiple distributions, you'll need to include each distribution's channel in your environment file.
 
 The weekly development builds of the QIIME 2 distributions can help you make sure your code stays current with the distribution(s) you are targeting as you can automate your testing against them.
 [](setup-dev-environment) will help you install the most recent successful development metapackage build (again, usually weekly, but sometimes the builds fail and take time to debug).
 
+There are a couple of things that we recommend implementing to help you ensure that your plugin remains compatible within the QIIME 2 ecosystem:
+
+1. Continuous Integration (CI) to regularly run your plugin's unit tests within the type of QIIME 2 environment where it should be used (i.e. Amplicon, Metagenome, Tiny plus custom selections).
+
+To implement this, you'll need to create a Github Action (GHA) that will be triggered each time you make a commit to your repository - either through a pull request (PR) or a direct commit to one of your remote branches.
```

Comment:
Should this be templated in, in place of the one we already have in the cookiecutter?

Reply:
I've been going back and forth about this, and I'm kind of leaning towards having them implement this manually vs. having it get automatically added when they create their plugin. I think it's more beneficial to have them set this up when their plugin is actually ready to be automatically tested vs. before they have any functionality laid out in their repository. I could totally be convinced otherwise, but I think it might help them understand this process a bit better if they have to set it up themselves (esp since they need to create a matching environment file that probably won't exist right away). I also worry that if they do have this setup too soon and the action is failing, it'll be more confusing for them.